tests/pyfunc/test_spark.py

```diff
@@ -1538,3 +1538,31 @@ def predict(self, context, model_input):
     pdf = df.toPandas()
     assert pdf["output"][0] == ("a", [0], (True, ("some_string",)), [(0.1,)])
     assert pdf["output"][1] == ("b", [1], (False, ("another_string",)), [(0.2,), (0.3,)])
+
+
+def test_spark_udf_env_manager_with_invalid_pythonpath(
+    spark, sklearn_model, model_path, monkeypatch
+):
+    origin_python_path = os.environ.get("PYTHONPATH", "")
+    monkeypatch.setenv("PYTHONPATH", f"{origin_python_path}:/tmp/abcdefg:/tmp/12345")
+    with mock.patch(
+        "mlflow.utils.databricks_utils.is_in_databricks_runtime", return_value=True
+    ), mock.patch("mlflow.utils.virtualenv._is_pyenv_available", return_value=True), mock.patch(
+        "mlflow.utils.virtualenv._get_pyenv_bin_path", return_value="pyenv"
+    ):
```

Comment:
why do we need this?

Reply:
`_remove_inaccesible_python_path` is only triggered if `is_in_databricks_runtime` is True, and _is_pyenv_available and _get_pyenv_bin_path have specific logic for databricks runtime

Reply:
> _is_pyenv_available and _get_pyenv_bin_path have specific logic for databricks runtime

Can you elaborate on this? Why do we need to pretend pyenv is available? Isn't that installed?

Reply:
`_is_pyenv_available` is defined like

```
def _is_pyenv_available():
    """
    Returns True if pyenv is available, otherwise False.
    """
    if is_in_databricks_runtime():
        return os.path.exists(_DATABRICKS_PYENV_BIN_PATH)
    else:
        return shutil.which("pyenv") is not None
```

in testing environment, we don't have `_DATABRICKS_PYENV_BIN_PATH` file, but in testing we mock `is_in_databricks_runtime` to return True, it causes route `return os.path.exists(_DATABRICKS_PYENV_BIN_PATH)` being executed.

`_get_pyenv_bin_path` has similar issue

Reply:
I see, not a blocker, but can we define an environment variable for specifying the pyenv path for testing?

Reply:
we can , it needs modify _is_pyenv_available code too. we can update it in follow-up PR

Reply:
can we default to `shutil.which("pyenv")` if `_DATABRICKS_PYENV_BIN_PATH` is not found?

Reply:
https://github.com/mlflow/mlflow/pull/12807